internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -606,6 +632,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, xr), errUpdateStatus)
 	}
 
+	if r.kindObserver != nil {
+		var gvks []schema.GroupVersionKind
+		for _, ref := range xr.GetResourceReferences() {
+			gvks = append(gvks, ref.GroupVersionKind())
+		}
+		r.kindObserver.WatchComposedResources(gvks...)
```

Comment:
if we now watch composed resources and trigger the reconcile in realtime, should we stop ending the loop with:

```golang
return reconcile.Result{RequeueAfter: r.pollInterval}, errors.Wrap(r.client.Status().Update(ctx, xr), errUpdateStatus)
```

and do instead:

```golang
return reconcile.Result{}, errors.Wrap(r.client.Status().Update(ctx, xr), errUpdateStatus)

```

Reply:
We can't I guess. 1. environments 2. functions relying on 60s poll.

Should we do everything towards removing poll? Yes, please. @negz

Reply:
if environment feature is not enabled and composition is not based on functions, then we could? Also, we could watch environment for changes using the same `WatchComposeResource` approach, but keep in memory relationships between environments and compositions, to know later what composition to reconcile.

Reply:
Am not against that direction, quite the opposite. Worth an attempt in a follow-up to see how far we get (in e2e).

For functions, I would like to see us being explicit that there is no 60s poll guarantee. Maybe we need an API to ask for one if necessary.

Reply:
That will be step 3: https://github.com/crossplane/crossplane/pull/4822